### DISCUSSION TITLE
deck.gl addControl equivalent?

### DISCUSSION THREAD

**Discussion by mmz15** [visgl/deck.gl#5448] [Q&A] [answered]
If I want to use ScaleControl, FullscreenControl, or GeolocateControl from mapboxgl, is that something I can do with deck.gl would I need to create from scratch? Said differently, is there a pattern to replace map.addControl?

For reference, here's a similar discussion for react-map-gl: https://github.com/visgl/react-map-gl/issues/197, and here's the documentation for the map.addControl API: https://docs.mapbox.com/mapbox-gl-js/api/markers/

**Comment by Pessimistress** [comment, marked as answer]
The issue you are referencing is 4 years old. If you are using React and Mapbox, you can use all react-map-gl controls: https://deck.gl/docs/api-reference/react/deckgl#react-context

If you are using vanilla JS with mapbox, you can use deck as a [mapbox custom layer](https://deck.gl/docs/api-reference/mapbox/overview) with all of the native controls.

**Reply by mmz15** [reply]
Super helpful, thanks for pointing that out! I'm still wrapping my head around the documentation. 

Appreciate your contributions!
